presto-hive/src/main/java/com/facebook/presto/hive/ColumnarBinaryHiveRecordCursor.java

```diff
@@ -321,7 +326,7 @@ public long getLong(int fieldId)
     {
         checkState(!closed, "Cursor is closed");
 
-        if (!types[fieldId].equals(BIGINT) && !types[fieldId].equals(DATE) && !types[fieldId].equals(TIMESTAMP)) {
+        if (!types[fieldId].equals(BIGINT) && !types[fieldId].equals(INTEGER) && !types[fieldId].equals(DATE) && !types[fieldId].equals(TIMESTAMP)) {
```

Comment:
Cannot we have it done similar to the approach from `ParquetPageSourceFactory` (with `Set.contains()`)? Or if performance concerns arise maybe we can at least extract this to a separate method (since you have the same check here and in `ColumnarTextHiveRecordCursor` and you definitely want it to be the same)?